pkg/disk/partition_table.go

```diff
@@ -204,6 +204,15 @@ type CustomPartitionTableOptions struct {
 	// filesystem if it is a supported type for that fileystem. If it is not,
 	// xfs is used as a fallback.
 	DefaultFSType FSType
+
+	// RequiredSizes defines a map of minimum sizes for specific directories.
+	// These indirectly control the minimum sizes of partitions. A directory
+	// with a required size will set the minimum size of the partition with the
+	// mountpoint that contains the directory. Additional directory
+	// requirements are additive, meaning the minimum size for a mountpoint's
+	// partition is the sum of all the required directory sizes it will
+	// contain.
+	RequiredSizes map[string]uint64
```

Comment:
nitpick: I'd call it `RequiredMinSizes` or maybe reuse `FilesystemCustomization` so that it's clear just from the variable name that these are minimums and not an exact size (even though the docstring clearly explains that).

Reply:
`RequiredMinSizes` sounds good.  I think originally "required" was meant to imply "lower bound"/"minimum requirements" sort of, but adding `Min` to the mix makes it clearer.

Reply:
Renamed.